### PR TITLE
feat: add qr scanner

### LINKED ISSUE #55: Implement `ScanQrPage`
Investigate using QR scanning to load up the recipient's DID into the `To:` field in `SendDidPage`. If it makes sense, integrate into didpay

Some notable libraries:
- https://pub.dev/packages/mobile_scanner
- https://pub.dev/packages/qr_code_scanner

### PR DESCRIPTION
**Pull request by ethan-tbd**
closes #55 

also generates a DID QR code in `AccountDidPage`

https://github.com/TBD54566975/didpay/assets/125412902/20fad73d-b7c5-43da-b0f3-d44463546195

simulator_screenshot_1E6940B2-08F0-42C9-86A8-50F1D4D2ACF2

### CODE REVIEW

**Comment by michaelneale** [comment]
nice!

File: lib/features/send/send_did_page.dart [hunk 1]
```diff
@@ -38,80 +58,145 @@ class SendDidPage extends HookWidget {
...
+  }
+
+  Future<bool> _isPhysicalDevice() async {
+    final deviceInfo = DeviceInfoPlugin();
+    return Platform.isIOS
+        ? (await deviceInfo.iosInfo).isPhysicalDevice
+        : Platform.isAndroid
+            ? (await deviceInfo.androidInfo).isPhysicalDevice
+            : false;
+  }
```
Comment (wesbillman) [inline lib/features/send/send_did_page.dart:166]: I would move this out to a `provider` somewhere so we can use it again when needed.

File: lib/features/send/send_did_page.dart [hunk 2]
```diff
@@ -38,80 +58,145 @@ class SendDidPage extends HookWidget {
...
+
+  void _scanQrCode(
+    BuildContext context,
+    TextEditingController controller,
+    ValueNotifier<String?> errorText,
+    String errorMessage,
+    String did,
+  ) async {
+    if (await _isPhysicalDevice()) {
```
[the remark this answers is not in the document]
Comment (wesbillman) [reply lib/features/send/send_did_page.dart:180]: We can chat about `use_build_context_synchronously` someday too ;)

File: lib/features/send/send_did_page.dart [hunk 3]
```diff
@@ -38,80 +58,145 @@ class SendDidPage extends HookWidget {
...
+  ) {
+    return Padding(
+      padding: const EdgeInsets.symmetric(vertical: Grid.xxs),
+      child: ListTile(
+        leading: const Icon(Icons.qr_code),
+        title: Text(
+          Loc.of(context).scanQrCode,
+          style: Theme.of(context).textTheme.bodyMedium,
+        ),
+        contentPadding: const EdgeInsets.symmetric(horizontal: Grid.side),
```
[the remark this answers is not in the document]
Comment (ethan-tbd) [reply lib/features/send/send_did_page.dart:151]: we use `20.0` for our side padding and `ListTile` defaults to `16.0`. will change that in the theme

File: lib/features/send/send_did_page.dart (hunk 2 above)
[the remark this answers is not in the document]
Comment (ethan-tbd) [reply lib/features/send/send_did_page.dart]: instead added a simulate function that splits out the if check in `onTap` of the `ListTile`

**Pull request merged**